src/details/desugar.cpp

```diff
@@ -0,0 +1,293 @@
+#include "ast.hpp"
+namespace minilua::details::ast {
+/**
+ * we have a for loop like this:
+ *      for v = e1, e2, e3 do block end
+ *
+ * and are basically deriving this code from it:
+ *
+ * do                                                                       --do_statement
+       local var, limit, step = tonumber(e1), tonumber(e2), tonumber(e3)    --statement_1
+       if not (var and limit and step) then error() end                     --statement_2
+       var = var - step                                                     --statement_3
+       while true do                                                        --while_loop
+         var = var + step                                                   --statement_w1
+         if (step >= 0 and var > limit) or (step < 0 and var < limit) then  --statement_w2
+           break
+         end
+         local v = var                                                      --statement_w3
+         block                                                              --all the statements of
+ the for loop end end
```

Comment:
The newline here makes it a bit hard to understand. Maybe wrap the entire "result example" in a code block (i.e. ```). Then clang format might ignore the long line (but not sure).